src/object/cli_obj.c

```diff
@@ -1848,6 +1849,19 @@ shard_result_process(tse_task_t *task, void *arg)
 	return 0;
 }
 
+static int
+shard_epoch_process(tse_task_t *task, void *arg)
+{
+	struct obj_auxi_args	*obj_auxi = (struct obj_auxi_args *)arg;
+	struct shard_auxi_args	*shard_auxi;
+
+	shard_auxi = tse_task_buf_embedded(task, sizeof(*shard_auxi));
+	if (*obj_auxi->epoch > shard_auxi->epoch && shard_auxi->epoch != 0)
+		*obj_auxi->epoch = shard_auxi->epoch;
```

Comment:
at API comments:
"* \param[in,out]
 *		epoch	[in]: The up boundary for the data to be synced.
 *			      DAOS_EPOCH_MAX means sync all.
 *			[out]: If the input one is DAOS_EPOCH_MAX, then the
 *			       output contains current highest epoch on the
 *			       (leader) server that has been synced. If the
 *			       object resides on multiple redundancy groups,
 *			       the lowest epoch that have been synced among
 *			       these RDGs  will be returned.
"
But here it will reset the user input epoch despite of weather it is DAOS_EPOCH_MAX, is it consistent?

Reply:
Now, I am thinking Niu's suggestion that whether we should allow the user to specify the @epoch input parameter, because if the object has multiple redundancy groups, then it is normal that each RDG has each own independent highest epoch. Under such case, the client given @epoch may be meaningless.